app/react/portainer/settings/SettingsView/SSLSettingsPanel/SSLSettingsPanel.tsx

```diff
@@ -0,0 +1,162 @@
+                        name="forceHTTPS"
+                        onChange={(value) => setFieldValue('forceHTTPS', value)}
+                      />
+                    </div>
+                  </div>
+
+                  <div className="form-group">
+                    <div className="col-sm-12">
+                      <TextTip color="blue">
+                        Provide a new SSL Certificate to replace the existing
+                        one that is used for HTTPS connections.
+                      </TextTip>
+                    </div>
+                  </div>
+
+                  <FormControl
+                    label="SSL/TLS certificate"
+                    tooltip="Select an X.509 certificate file, commonly a crt, cer or pem file."
+                    inputId="ca-cert-field"
+                    errors={errors.certFile}
+                  >
+                    <FileUploadField
+                      required={typeof errors.certFile !== 'undefined'}
+                      inputId="ca-cert-field"
+                      name="certFile"
+                      onChange={(file) => setFieldValue('certFile', file)}
+                      value={values.certFile}
+                    />
+                  </FormControl>
+
+                  <FormControl
+                    label="SSL/TLS private key"
+                    tooltip="Select a private key file, commonly a key, or pem file."
+                    inputId="ca-cert-field"
+                    errors={errors.keyFile}
+                  >
+                    <FileUploadField
+                      required={typeof errors.keyFile !== 'undefined'}
+                      inputId="ca-cert-field"
+                      name="keyFile"
+                      onChange={(file) => setFieldValue('keyFile', file)}
+                      value={values.keyFile}
+                    />
+                  </FormControl>
+
+                  <div className="form-group">
+                    <div className="col-sm-12">
+                      <LoadingButton
+                        isLoading={mutation.isLoading || reloadingPage}
+                        disabled={!isValid}
+                        loadingText={reloadingPage ? 'Reloading' : 'Saving'}
+                        className="!ml-0"
+                      >
+                        Save SSL Settings
+                      </LoadingButton>
+                    </div>
+                  </div>
+                </Form>
+              )}
+            </Formik>
+          </Widget.Body>
+        </Widget>
+      </div>
+    </div>
+  );
+
+  function handleSubmit({ certFile, forceHTTPS, keyFile }: FormValues) {
+    if (!certFile || !keyFile) {
+      return;
+    }
+
+    mutation.mutate(
+      { certFile, httpEnabled: !forceHTTPS, keyFile },
+      {
+        async onSuccess() {
+          await new Promise((resolve) => {
+            setTimeout(resolve, 5000);
+          });
+          window.location.reload();
```

Comment:
yes. since ssl certificates are reloaded, the server is restarted and new to make sure the app is connected.

we could potentially use something like the update loading thing. I'll have a look

Reply:
I'll just leave the current behavior.